geomstats/geometry/special_euclidean.py

```diff
@@ -1108,6 +1123,28 @@ def parallel_transport(
         return homogeneous_representation(
             transported_rot, translation, max_shape, 0.)
 
+    def _private_squared_dist(self, point_a, point_b):
+        dist = super().squared_dist(point_a, point_b)
+        return dist
+
+    def squared_dist(self, point_a, point_b):
+        """Squared geodesic distance between two points.
+
+        Parameters
+        ----------
+        point_a : array-like, shape=[..., dim]
+            Point.
+        point_b : array-like, shape=[..., dim]
+            Point.
+
+        Returns
+        -------
+        sq_dist : array-like, shape=[...,]
+            Squared distance.
+        """
+        dist = _squared_dist(point_a, point_b, metric=self)
```

Comment:
I didn' t understand why we have `squared_dist`, `_squared_dist` and `_private_squared_dist`.
Also, if this works, this could go into the parent class (even into `RiemannianMetric`)? I put it here before to test it on SE(3) but this is more general.

Reply:
Yes it's a hack that I found for the following reasons.

- Reasoning behind `_private_squared_dist`: 

When I remove _private_squared_dist and put `dist = _squared_dist(point_a, point_b, metric=super())` on this line, there is an error in tensorflow, as in: the gradient becomes accurate to only 1e-1 precision (?).

- Reasoning behind `_squared_dist()`:

The _squared_dist() is there to put the metric as the last parameter, for it to be a kwarg instead of an arg, for it to work with tensorflow.

I'll have another try at it and put all of this in Riemannian metric indeed.